src/main/java/net/minecraftforge/fml/common/discovery/ASMDataTable.java

```diff
@@ -114,11 +114,16 @@ public boolean apply(ASMData data)
         if (containerAnnotationData == null)
         {
             ImmutableMap.Builder<ModContainer, SetMultimap<String, ASMData>> mapBuilder = ImmutableMap.builder();
-            for (ModContainer cont : containers)
+            //concurrently filter the values to speed this up
+            containers.parallelStream().forEach(cont ->
```

Comment:
If we're using a stream here, we could probably do the whole thing with it. Something like:
```suggestion
            containers.parallelStream()
                      .map(cont -> Pair.of(cont, ImmutableSetMultimap.copyOf(Multimaps.filterValues(globalAnnotationData, new ModContainerPredicate(cont)))))
                      .collect(ImmutableMap.toImmutableMap(pair -> pair.getKey(), pair-> pair.getValue()))
```